app/src/main/java/com/android/solvit/provider/ui/request/ListRequestsFeed.kt

```diff
@@ -0,0 +1,434 @@
+              error = painterResource(id = R.drawable.error),
+              contentDescription = "Service Image",
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .height(160.dp)
+                      .border(1.dp, Color.White, RoundedCornerShape(12.dp))
+                      .clip(RoundedCornerShape(12.dp)),
+              contentScale = ContentScale.Crop)
+        } else {
+          Box(
+              modifier =
+                  Modifier.fillMaxWidth()
+                      .height(160.dp)
+                      .border(1.dp, Color.White, RoundedCornerShape(12.dp))
+                      .clip(RoundedCornerShape(12.dp)),
+              contentAlignment = Alignment.Center) {
+                Text(text = "No Image Provided", fontSize = 16.sp, color = Color.White)
+              }
+        }
+        Spacer(modifier = Modifier.height(8.dp))
+        Row(horizontalArrangement = Arrangement.SpaceBetween, modifier = Modifier.fillMaxWidth()) {
+          InteractionBar("Comment", R.drawable.comment_icon, onClick)
+          InteractionBar("Share", R.drawable.share_icon, onClick)
+          InteractionBar("Reply", R.drawable.reply_icon, onClick)
+        }
+        selectedLocation?.let { GetDirectionsBubble(location = it) { selectedLocation = null } }
+      }
+}
+
+@Composable
+fun InteractionBar(text: String, icon: Int, onClick: () -> Unit = {}) {
+  Column(horizontalAlignment = Alignment.CenterHorizontally) {
+    Text(text = text, fontSize = 14.sp, color = Color.White)
+    IconButton(onClick = { onClick() }) {
+      Image(
+          painter = painterResource(id = icon),
+          contentDescription = text,
+          colorFilter = ColorFilter.tint(Color.White),
+          modifier = Modifier.size(21.dp))
+    }
+  }
+}
+
+@Composable
+fun FilterBar(
+    selectedService: String,
+    selectedFilters: Set<String>,
+    onSelectedService: (String) -> Unit,
+    onFilterChange: (String, Boolean) -> Unit
+) {
+  val filters = listOf("Service", "Near Me", "Due Time")
+
+  LazyRow(
+      modifier = Modifier.testTag("FilterBar"),
+      horizontalArrangement = Arrangement.spacedBy(5.dp),
+      verticalAlignment = Alignment.Top) {
+        items(filters.size) { idx ->
+          val filter = filters[idx]
+          if (filter == "Service") {
+            ServiceChip(
+                selectedService = selectedService,
+                onServiceSelected = { service ->
+                  onSelectedService(service)
+                  onFilterChange("Service", true)
+                })
+          } else {
+            FilterChip(
+                label = filter,
+                isSelected = selectedFilters.contains(filter),
+                onSelected = { isSelected -> onFilterChange(filter, isSelected) })
+          }
+        }
+      }
+}
+
+@Composable
+fun ServiceChip(selectedService: String, onServiceSelected: (String) -> Unit) {
```

Comment:
Nit: There is no clear indication that this Chip is a dropdown menu. Could an icon or a question mark be added to show the call for action?

Reply:
I implemented this and will add it to the PR before merging it, thanks!